rules-tests/TypeDeclaration/Rector/FunctionLike/ReturnTypeDeclarationRector/Fixture/skip_inferred_this.php.inc

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace T1 {
+    class A {
+        /**
+         * @return $this
+         */
+        public function itself()
+        {
+        }
```

Comment:
Here the return type is `void`, so the behavior is correct. The docblock is invalid.

Reply:
This is correct, yes. The problem is in the other return type. It doesn't change if I specify the return type for `itself` as `static` and add a `return $this;`. If I remove either of those, the return type of `B::test()` changes, which is definitely a problem

Reply:
The provided code is invalid: https://3v4l.org/Pa9mu
Rector can work only with valid PHP code.

Reply:
I've changed it to `return $this;`. The problem persists.